main/src/library/Array.flix

```diff
@@ -1545,13 +1546,29 @@ mod Array {
             forWithAccum(lo+1, hi, ac1, f)
         }
 
+    ///
+    /// Copies the range between `srcPos` (inclusive) to `srcPos + len` (exclusive) of `src` into `dst` starting from `dstPos` (inclusive).
+    /// A valid range has the following properties:
+    ///   0 <= len.
+    ///   0 <= `srcPos`.
+    ///   0 <= `dstPos`.
+    ///   `srcPos + len` <= `length(src)`.
+    ///   `dstPos + len` <= `length(dst)`.
+    ///
+    /// If `src` and `dst` refer to the same array the result will be as though the range was copied to a separate array before pasting to the `dst` array.
```

Comment:
What happens if it's out of range?

Reply:
Probably exception (un tracked).

What do you propose? I think copyArray also has no special logic does it?

Reply:
I think thats reasonable. @mlutze What did you have in mind?

Reply:
I assumed that was the case. I just thought some documentation of that would be good.

Reply:
Yeah, but we don't have a mechanism to document Java exceptions (yet).